docs/03-deploy.Rmd

```diff
@@ -10,9 +10,9 @@ If you do not mind using command-line tools or are familiar with GIT/GitHub, you
+As we just mentioned, Netlify allows you to quickly publish a website by uploading the `public/` folder through its web interface, and you will be assigned a random subdomain `*.netlify.com`^[You don't have to keep the `*.netlify.com` domain. See Appendix \@ref(domain-name) for more information.]. This approach is good for those websites that are not updated frequently (or at all). However, it is unlikely that you do not need to update your website, so we introduce a better approach in this section,^[Please bear in mind that the purpose of this section is to outline the basic steps of publishing a website with Netlify, and the technical details may change from time to time, so the official Netlify documentation should be the most reliable source if you have any questions or anything we introduced here does not work.] which will take you a few more minutes to complete the configurations. Once it is properly configured, all you need to do in the future is to update the source repository, and Netlify will call Hugo to render your website automatically.
 
-Basically you have to host all source files of your website in a GIT repository. You do not need to put the `public/` directory under version control^[You can add `public` to `.gitignore` to ignore it in GIT.] because it will be automatically generated. Currently Netlify supports GIT repositories hosted on GitHub, GitLab, and BitBucket. With any of these accounts, you can log into Netlify from its homepage and follow the guide to create a new site from your GIT repository.
+Basically you have to host all source files of your website in a GIT repository^[The GIT repository needs to be your root directory for Netlify to properly build your site. If your GIT repository contains a directory which then contains your site's contents, Netlify will not build.]. You do not need to put the `public/` directory under version control^[You can add `public` to `.gitignore` to ignore it in GIT.] because it will be automatically generated. Currently Netlify supports GIT repositories hosted on GitHub, GitLab, and BitBucket. With any of these accounts, you can log into Netlify from its homepage and follow the guide to create a new site from your GIT repository.
```

Comment:
> If your GIT repository contains a directory which then contains your site's contents, Netlify will not build.

Actually if the repo is just the `public/` directory, you can still use Netlify to deploy it. You just don't specify any build command (leave it empty), and specify the publish directory to be `.` (root directory). Netlify will not run Hugo or whatever generator to build you site, but simply deploys the repo as is.

Users probably won't discover this, so I need your help to explain it.

This is also why there are no advantages for Hugo users to use Github Pages. Github Pages uses the token `.nojekyll` if your website has been prebuilt so that it won't run `jekyll` again. For Netlify, just don't specify the build command.

Reply:
The issue I ran into here wasn't that I had the repo as just the `public/` directory, but the opposite. I had `Repo/Blogdown_Contents`. That is, all of the contents of my blogdown site were within another directory instead of in the root directory. Netlify wouldn't build without all of the necessary contents being within the root directory. I suppose it could be reworded as something like:

> If the contents of your blogdown site are not in the root directory of your GIT repository, Netlify will not build.

Reply:
I see. In that case, you can specify the build command to be `cd Blogdown_Contents; hugo`. The build command can be arbitrary shell commands.